include/experimental/__p1673_bits/blas1_vector_norm2.hpp

```diff
@@ -56,37 +57,19 @@ template<class in_vector_t,
 void vector_norm2(in_vector_t x,
                   Scalar& result)
 {
-  using std::abs;
   using std::sqrt;
 
-  // Rescaling, as in the Reference BLAS DNRM2 implementation, avoids
-  // unwarranted overflow or underflow.
+  // Initialize the sum of squares result
+  sum_of_squares_result<Scalar> ssq_init;
```

Comment:
If `Scalar` is a floating-point type (`float`, `double`, or `long double`), you can do this in one line:
```
auto result = ssq_res = vector_sum_of_squares(x, {Scalar{}, Scalar{1}});
```
However, this brings up the interesting problem of what to do if `Scalar` is _not_ a floating-point type.  For instance, `Scalar` may be some type that doesn't know how to be initialized from `int` or `double`.  I think we can do this via `if constexpr`:
```
if constexpr(std::is_floating_point_v<Scalar>) {
  const auto ssq_res = vector_sum_of_squares(x, {Scalar{}, Scalar{1}});
  return ssq_res.scaling_factor * sqrt(ssq_res.scaled_sum_of_squares);
} else {
  Scalar result{};
  for(size_t k = 0; k < x.extent(0); ++k) {
    result += abs(x(i)) * abs(x(i));
  }
  return sqrt(result);
}
```